plugins/plugin-logui/src/controller/kubectl/logs.ts

```diff
@@ -14,56 +14,27 @@
  * limitations under the License.
  */
 
-import { Abortable, Arguments, Registrar } from '@kui-shell/core'
-import {
-  KubeOptions,
-  doExecWithPty,
-  doExecWithStdout,
-  defaultFlags as flags,
-  getNamespace,
-  hasLabel
-} from '@kui-shell/plugin-kubeui'
+import Debug from 'debug'
+import PrettyPrintAnsiString from 'ansi_up'
+import * as colors from 'colors/safe'
+import { Abortable, Arguments, Registrar, Streamable } from '@kui-shell/core'
+import { KubeOptions, doExecWithPty, defaultFlags as flags, isHelpRequest } from '@kui-shell/plugin-kubeui'
 
 import commandPrefix from '../command-prefix'
-import { formatAsTable } from '../../renderers/table'
 
 interface LogOptions extends KubeOptions {
   f: string
   follow: string
   previous: boolean
+  tail: number
 }
 
-/**
- * don't fetch too many log entries
- *
- */
-function prepareCommand(args: Arguments<LogOptions>): string {
-  //
-  // the default log limit is... unlimited? let's make sure we don't
-  // cause chaos here by requesting too many log lines
-  //
-  let command = args.command.replace(new RegExp(`^\\s*${commandPrefix}`), '')
-
-  if (!args.parsedOptions.tail && !args.parsedOptions.since) {
-    // debug('limiting log lines')
-    command = `${command} --tail 20`
-  }
-
-  return command
-}
-
-/**
- * Fetch an interval of log records, and display them as a Kui table.
- *
- */
-async function getLogsAsTable(args: Arguments<LogOptions>) {
-  const raw = await doExecWithStdout(args, prepareCommand)
+const debug = Debug('plugin-logui/controller/kubectl/logs')
 
-  return formatAsTable(raw, {
-    name: args.argvNoOptions[args.argvNoOptions.indexOf('logs') + 1],
-    namespace: (args && getNamespace(args)) || 'default'
-  })
-}
+const literal = (match, p1, p2) => `${p1}${colors.blue(p2)}`
```

Comment:
all these color are hard coded? will it be affected by themes

Reply:
these are using ansi control codes. the theme alignment comes from plugin-kubeui-client/web/css/colors.css